CTOR-1053 : Refresh cloud-aws-cloudwatchlogs (#3728)
centreon/centreon-documentation: pp/integrations/plugin-packs/procedures/cloud-aws-cloudwatchlogs.md

import TabItem from '@theme/TabItem';

CloudWatch Logs enables you to centralize the logs from all of your systems, applications, 
and AWS services that you use, in a single, highly scalable service. You can then easily view them, 
search them for specific error codes or patterns, filter them based on specific fields, 
or archive them securely for future analysis.

> > This Plugin may generate very huge queries toward CloudWatch Logs API. Make sure to use filtering capability

Reviewer: ```suggestion
> This plugin may generate very huge queries toward the CloudWatch Logs API. Make sure you use the plugin's filtering capabilities
```

> > with ```--group-name``` and ```--stream-name``` options. 

Reviewer: ```suggestion
> using the```--group-name``` and ```--stream-name``` options. 
```

## Connector dependencies

The following monitoring connectors will be installed when you install the **Amazon CloudWatch Logs** connector through the
**Configuration > Monitoring Connector Manager** menu:
* [Base Pack](./base-generic.md)

## Pack assets

### Templates

The Monitoring Connector **Amazon CloudWatch Logs** brings a host template:

* **Cloud-Aws-CloudWatchLogs-custom**

The connector brings the following service templates (sorted by the host template they are attached to):

<Tabs groupId="sync">
<TabItem value="Cloud-Aws-CloudWatchLogs-custom" label="Cloud-Aws-CloudWatchLogs-custom">

> There is no service template linked to this host template.

Reviewer: ```suggestion
No service template is linked to this host template.
```

```

</TabItem>
<TabItem value="Debian 11 & 12" label="Debian 11 & 12">

```bash
apt install centreon-plugin-cloud-aws-cloudwatchlogs-api
```

</TabItem>
<TabItem value="CentOS 7" label="CentOS 7">

```bash
yum install centreon-plugin-Cloud-Aws-Cloudwatchlogs-Api
```

</TabItem>
</Tabs>

## Using the monitoring connector

### Using a host template provided by the connector

1. Log into Centreon and add a new host through **Configuration > Hosts**.
2. Fill in the **Name**, **Alias** & **IP Address/DNS** fields according to your resource's settings.
3. Apply the **Cloud-Aws-CloudWatchLogs-custom** template to the host. A list of macros appears. Macros allow you to define how the connector will connect to the resource, and to customize the connector's behavior.
4. Fill in the macros you want. Some macros are mandatory.

| Macro         | Description                                                                                                                | Default value     | Mandatory   |
|:--------------|:---------------------------------------------------------------------------------------------------------------------------|:------------------|:-----------:|

> | AWSACCESSKEY  | Set AWS access key                                                                                                         |                   |             |
> | AWSCUSTOMMODE | When a plugin offers several ways (CLI, library, etc.) to get information the desired one must be defined with this option | awscli            |             |
> | AWSREGION     | Set the region name (required)                                                                                             |                   |             |
> | AWSSECRETKEY  | Set AWS secret key                                                                                                         |                   |             |

Author: ```suggestion
| AWSACCESSKEY  | Set AWS access key                                                                                                         |                   | X           |
| AWSCUSTOMMODE | When a plugin offers several ways (CLI, library, etc.) to get information the desired one must be defined with this option | awscli            |             |
| AWSREGION     | Set the region name (required)                                                                                             |                   | X           |
| AWSSECRETKEY  | Set AWS secret key                                                                                                         |                   | X           |
```

| EXTRAOPTIONS   | Any extra option you may want to add to the command (a --verbose flag for example). All options are listed [here](#available-options).                 | --verbose         |             |

</TabItem>
</Tabs>

3. [Deploy the configuration](/docs/monitoring/monitoring-servers/deploying-a-configuration). The service appears in the list of services, and on the **Resources Status** page. The command that is sent by the connector is displayed in the details panel of the service: it shows the values of the macros.

## How to check in the CLI that the configuration is OK and what are the main options for?

Once the plugin is installed, log into your Centreon poller's CLI using the
**centreon-engine** user account (`su - centreon-engine`). Test that the connector 
is able to monitor an AWS Instance using a command like this one (replace the sample values by yours):

```bash
/usr/lib/centreon/plugins/centreon_aws_cloudwatchlogs_api.pl \
	--plugin=cloud::aws::cloudwatchlogs::plugin \
	--mode=get-logs \
	--custommode='awscli' \

> 	--aws-secret-key='' \
> 	--aws-access-key='' \
> 	--aws-role-arn='' \
> 	--proxyurl=''  \
> 	--region='' \

Author: ```suggestion
	--aws-secret-key='XXXX' \
	--aws-access-key='XXXX' \
	--aws-role-arn='' \
	--proxyurl=''  \
	--region='eu-west-1' \
```

	--group-name='' \
	--stream-name='' \
	--start-time-since='' \
	--unknown-status='' \
	--warning-status='' \
	--critical-status='' \

> 	--verbose

Reviewer: the previous example was better
